rn/Teacher/src/modules/submissions/list/get-submissions-props.js

```diff
@@ -142,45 +128,79 @@ export function pendingProp (assignmentContent?: AssignmentContentState, courseC
+  let group = groupValues
+    .filter(({ group }) => group && group.group_category_id === categoryID)
+    .find(({ group }) => group.users && group.users.map(({ id }) => id).includes(submission.user_id))
+
+  if (!group) return
+  return group.group
+}
+
+export function groupPropsForSubmissionsAndDueDate (submission: SubmissionWithHistory, dueDate: ?string, group: ?Group) {
+  if (!group) return
+  return {
+    // just a precaution... we're filtering out empty groups below
+    userID: group.users && group.users.length > 0 ? group.users[0].id : 'none',
+    groupID: group.id,
+    name: group.name,
+    status: statusProp(submission, dueDate),
+    grade: gradeProp(submission),
+    score: submission ? submission.score : null,
+    submissionID: submission ? submission.id : null,
+    submission,
+    sectionID: null,
+    allSectionIDs: null,
+  }
+}
+
 export function getSubmissionsProps (entities: Entities, courseID: string, assignmentID: string): AsyncSubmissionsDataProps {
   // enrollments
   const courseContent = entities.courses[courseID]
   const enrollments = getEnrollments(courseContent, entities.enrollments)
 
-  const sectionIDs = enrollments.reduce((memo, enrollment) => {
+  const sectionIDs: { [string]: [string] } = enrollments.reduce((memo, enrollment) => {
     const userID = enrollment.user_id
     const existing = memo[userID] || []
     return { ...memo, [userID]: [...existing, enrollment.course_section_id] }
   }, {})
 
   // submissions
   const assignmentContent = entities.assignments[assignmentID]
-  const submissionsByUserID = getSubmissionsByUserID(assignmentContent, entities.submissions)
-
-  // don't create the submissions unless we have submissions
-  // this fixes some issues we were having where deep linking
-  // wouldn't show the submission for the user
-  const submissions = Object.keys(submissionsByUserID).length
-    ? uniqueEnrollments(enrollments)
-      .filter(e => e.type === 'StudentEnrollment')
-      .sort((e1, e2) => {
-        if (e1.type !== e2.type) {
-          if (e1.type === 'StudentEnrollment') {
-            return -1
-          } else if (e2.type === 'StudentEnrollment') {
-            return 1
-          }
+  const submissions = Object.values(entities.submissions)
+    // $FlowFixMe
+    .map(({ submission }) => submission)
+    .filter(sub => sub.assignment_id === assignmentID)
+    .map(sub => {
+      let group = getGroup(entities.groups, sub, assignmentContent.data.group_category_id)
+      if (assignmentContent.data.group_category_id != null &&
+          !assignmentContent.data.grade_group_students_individually &&
+          group != null) {
+        const due = assignmentContent != null ? dueDate(assignmentContent.data, null, group) : undefined
+        return groupPropsForSubmissionsAndDueDate(sub, due, group)
+      } else {
+        const enrollment = enrollments.find(e => e.user_id === sub.user_id)
+        if (!enrollment) {
+          return
```

Comment:
We no longer need to do the song and dance of matching user enrollments up to potentially missing submissions since we now are guaranteed to have one submission per user/group. We also handle both groups and users here since it's possible for a user to not be in a group so we can't do one or the other, we must do both.